### PR TITLE
Add disputeTransaction spec

### PR DESCRIPTION
**Pull request by tubaxenor**

### CODE REVIEW

**Comment by tubaxenor** [comment]
@ngan, this is good to review

File: test/disputeTransaction.js [hunk 1]
```diff
@@ -1,23 +1,148 @@
...
 
 contract("InkProtocol", (accounts) => {
-  let protocol
-
-  beforeEach(async () => {
-    protocol = await InkProtocol.new()
-  })
+  buyer = accounts[1]
+  seller = accounts[2]
+  unknown = accounts[accounts.length - 1]
```
Comment (ngan) [inline test/disputeTransaction.js]: Missing `let`.

File: test/disputeTransaction.js [hunk 2]
```diff
@@ -1,23 +1,148 @@
 const $util = require("./util")
 const InkProtocol = artifacts.require("./mocks/InkProtocolMock.sol")
+const ErroredPolicy = artifacts.require("./mocks/ErrorPolicyMock.sol")
```
Comment (ngan) [inline test/disputeTransaction.js]: `ErrorPolicy`

File: test/disputeTransaction.js [hunk 3]
```diff
@@ -1,23 +1,148 @@
...
+        policy,
+        protocol,
+        transaction
+      } = await $util.buildTransaction(buyer, seller, {
+        finalState: $util.states.Accepted
+      })
+
+      let fulfillmentExpiry = await policy.fulfillmentExpiry()
+      // 10 mins before expiry
+      $util.advanceTime(fulfillmentExpiry.toNumber() - 600)
```
Comment (ngan) [inline test/disputeTransaction.js]: Let's go with 10 seconds.

File: test/disputeTransaction.js [hunk 4]
```diff
@@ -1,23 +1,148 @@
...
+        protocol,
+        transaction
+      } = await $util.buildTransaction(buyer, seller, {
+        finalState: $util.states.Accepted
+      })
+
+      // 7 days for fulfillmentExpiry
+      $util.advanceTime(86400 * 7)
+
+      await protocol.disputeTransaction(transaction.id, {from: buyer})
```
Comment (ngan) [inline test/disputeTransaction.js]: Let's add space inside `{...}`... so `{ from: buyer }`

File: test/disputeTransaction.js [hunk 5]
```diff
@@ -1,23 +1,148 @@
...
+      let {
+        policy,
+        protocol,
+        transaction
+      } = await $util.buildTransaction(buyer, seller, {
+        finalState: $util.states.Accepted
+      })
+
+      let fulfillmentExpiry = await policy.fulfillmentExpiry()
+
```
Comment (ngan) [inline test/disputeTransaction.js]: Remove empty line here.

**Review by ngan — CHANGES_REQUESTED** [review]

**Comment by tubaxenor** [comment]
@ngan this is good to review

**Pull request merged**